lib/workspace.js

```diff
@@ -123,6 +123,9 @@ module.exports = {
             if (state == 'uninitialized') {
                 await module.exports.initialize(workspace_id);
                 await module.exports.updateState(workspace_id, 'stopped');
+            } else {
+                await module.exports.controlContainer(workspace_id, 'destroy');
```

Comment:
I think we can't send this message here, because the host might no longer exist at this point (it could be an old host). I think it would be more robust to have the running hosts check the DB to see whether any of their containers are `stopped` and delete them if so.

Reply:
So the concern here is that there might be nothing to destroy?

Reply:
I mean, I can totally remove these lines and rely on whatever method we come up with later to recycle those resources. I just put it there so whenever a user clicks the upload destroy the container and delete the folder on the host.